app/controllers/course/assessment/question/programming_controller.rb

```diff
@@ -45,6 +45,7 @@ def update
                                        try(:[], :skill_ids)
       @programming_question.assign_attributes(programming_question_params.
                                               except(:question_assessment))
+      @programming_question.is_synced_with_codaveri = false
```

Comment:
minor nit: this actually sets the sync to false even when we adjust non-codaveri related fields (e.g. staff comments)

Reply:
with Codaveri, almost all parts of each programming question needs to be synced with Codaveri

- title
- description
- language (name and version)
- templates (solutions)
- all test cases (public, private, evaluation) -> each has expression, expected, and hint

introducing logic to check if any one of those fields have changed will complicate the logic and makes the code very messy, hence it should be in our best interest that everytime we update the programming question, we mark it as not synced so that later when we really need to sync (getting live feedback, evaluating or generating codaveri feedback) we sync it properly and on-demand